foreach.h:14:34: error: declaration of namespace ‘boost::BOOST_FOREACH’ conflicts...

[  0%] Building CXX object opencog/util/CMakeFiles/util.dir/lazy_selector.cc.o
In file included from /home/dhart/opencog/opencog/util/lazy_selector.cc:28:0:
/home/dhart/opencog/opencog/util/foreach.h:14:34: error: declaration of namespace ‘boost::BOOST_FOREACH’ conflicts with
In file included from /usr/include/boost/multi_index/hashed_index.hpp:22:0,
                 from /home/dhart/opencog/opencog/util/lazy_selector.h:30,
                 from /home/dhart/opencog/opencog/util/lazy_selector.cc:25:
/usr/include/boost/foreach_fwd.hpp:56:1: error: previous declaration of namespace ‘boost::BOOST_FOREACH’ here
make[3]: **\* [opencog/util/CMakeFiles/util.dir/lazy_selector.cc.o] Error 1
make[2]: **\* [opencog/util/CMakeFiles/util.dir/all] Error 2
make[1]: **\* [opencog/learning/moses/CMakeFiles/moses.dir/rule] Error 2
make: **\* [moses] Error 2
(beefymiracle-virgin)dhart@opencog-aristotle:~/opencog_build$

--- a/opencog/util/foreach.h
+++ b/opencog/util/foreach.h
@@ -10,7 +10,7 @@ namespace boost {
 // Work-around for bug, see
 // https://svn.boost.org/trac/boost/ticket/6131  for details.
 // See also https://bugs.launchpad.net/opencog/+bug/1057640
-#if BOOST_VERSION != 104900
+#if (BOOST_VERSION != 104800) || (BOOST_VERSION != 104900)
 namespace BOOST_FOREACH = foreach;
 #endif
 

